### ISSUE TITLE
99% in CV_results but 50% with scikit-learn

### ISSUE THREAD

**Issue by koskr** [automl/auto-sklearn#751]
Good evening; 

I have used AutoSklearn for a binary classification of tabular data and got 0.99 as best validation score. 
When I read cv_results, there are numerous pipelines that indeed have given such a score. 
However, when I explicitely implement one of them with scikit-learn (say passive_aggressive with the same parameters as in CV_results) and train it, I get only 0.5 ! 

Why ? Is there something I do no get with those 'Best validation score' and 'Mean score' ?

It is very important. There is almost 3 monthes of work at stake. 

Thank you.

**Comment by mfeurer** [comment]
I can see two possible issues here:

1. a mistake in your explicit implementation. Could you please paste the final ensemble and your re-implementation here?
2. Overfitting to the validation set. Counter-measures would be very much dependent on your dataset and I'd need some more information.

**Comment by codloverS2** [comment]
I have  same problem. The parameters I used are:
`       classifier  =  autosklearn.classification.AutoSklearnClassifier(
        time_left_for_this_task=60*10,
        ml_memory_limit=10240,
        per_run_time_limit=60*6,
        include_estimators=['libsvm_svc'],
        include_preprocessors=["no_preprocessing"],
        ensemble_size=0,
        resampling_strategy='holdout',
        resampling_strategy_arguments={'train_size': 0.67},
        n_jobs=2,
        delete_tmp_folder_after_terminate= True,
        delete_output_folder_after_terminate = True
    )`

output: [(1.000000, SimpleClassificationPipeline({'balancing:strategy': 'weighting', 'categorical_encoding:__choice__': 'one_hot_encoding', 'classifier:__choice__': 'libsvm_svc', 'imputation:strategy': 'median', 'preprocessor:__choice__': 'no_preprocessing', 'rescaling:__choice__': 'quantile_transformer', 'categorical_encoding:one_hot_encoding:use_minimum_fraction': 'False', 'classifier:libsvm_svc:C': 558.8232074758778, 'classifier:libsvm_svc:gamma': 0.04911239226110395, 'classifier:libsvm_svc:kernel': 'rbf', 'classifier:libsvm_svc:max_iter': -1, 'classifier:libsvm_svc:shrinking': 'False', 'classifier:libsvm_svc:tol': 0.0001338916518519931, 'rescaling:quantile_transformer:n_quantiles': 446, 'rescaling:quantile_transformer:output_distribution': 'uniform'}

My  re-implementation:

`classifier = sklearn.svm.SVC(C=558.8232074758778,kernel = 'rbf',gamma = 0.04911239226110395, shrinking = False, tol = 0.0001338916518519931,  max_iter = -1, probability= True)
classifier.fit(X_train_svc,y_train_svc `)`

This is the way used to separate dataset:
`X_train_svc, X_test_svc, y_train_svc, y_test_svc = train_test_split(x, y, test_size=0.33, random_state=42)`

My task is text classification and got : 73% of accuracy, but if I use the same hyperparameters in SVM model, I got 53%. Why? I used the same proportion the data (train:67% and validation:33%). 

Thank you.

**Comment by mfeurer** [comment]
Given the detailed description of @MirelleB, I think I know what's going on. The reason for the dropped performance is because Auto-sklearn performs quite a lot of prerpocessing before actually calling the classifier. In your case this is actually using a weighting of the classes, one hot encoding and a quantile transformer to scale the input data. I'm afraid that the classifier alone will not work, and would suggest that you either rebuild the whole pipeline with sklearn modules or use pipeline from Auto-sklearn directly.

**Comment by MirelleB** [comment]
@mfeurer  Thanks for your  answer!
I created this pipeline:

`check_params = { 'libSVM__C':[0.1,0.5,1,500,1000], 'libSVM__gamma':[0.001,0.005,0.01,1,50]}
`

`pipeline = Pipeline(steps=[
    ('StandardScaler',RobustScaler()),
    ('pca',PCA()),
    ('libSVM',sklearn.svm.SVC(kernel = 'rbf', cache_size = 1000,class_weight='balanced',max_iter = -1, probability= True))
])
`

`model = GridSearchCV(pipeline, check_params, cv = 5, n_jobs=20)`

`model.fit(x,y)`

and my acurracy stayed within the expected value.

Thanks very much.

**Comment by mfeurer** [comment]
Cool. I guess I'm closing this for now. Please reopen or comment if there are further questions on this topic.

**Comment by AndoStat** [comment]
I have tested sklearn significantly on validating and testing real prediction for known data for various HP systematic analytically from 1 to 200 neurons in 1 to 7  HL. Test shoved that in rare cases i get 65%+ newer reaching 70 %. Tested within 4 activation function and 3 different solvers, used MLP NN. Sklearn MLP neural network in this case is a pure waste of time and is data noise generator. Tested on test files and validation data files. Also u cant a play HP discovered within one flow app to another, u get poor AUC instantly. If software used does not have HP estimator integrated (that's always the case with scikit-learn) u re out of luck, determining HP manually ... more than a jackpot. I'm using MatLab now, IBM SPSS also not bad. For instance SPSS auto HP determination finds out neurons in just one layer with high AUC on test and validation datasets, that's something sklearn doesn't reach in my test flows on pure tabular numerical dataset.
